resources/views/search.blade.php

```diff
@@ -2,42 +2,122 @@
+            <span class="h3">&nbsp;<span class="badge badge-default">{{$count}} items</span></span>
+         @else
+            <span class="h3">&nbsp;<span class="badge badge-default">{{ count($products) }} items</span></span>
+         @endif
+   	</ol>
+   	<!-- END BREADCRUMB -->
+   </div>
+   <!-- End Row -->
+
+   <!-- Row Category Search-->
+   <div class="row-fluid pt-3">
+      <?php $categoriesFound = array(); ?>
+      @if( !isset($category) )
+         <!-- Category Buttons -->
+         <span style="font-size:1.4em;">Refine your search:</span>
+         @foreach ($products as $product)
+            @if(!in_array($product->category->name,$categoriesFound))
+               <a role="button" class="btn btn-secondary" href="{{$searchWord}}/{{$product->category->id}}">{{ $product->category->name }}</a>
+            @endif
+            <?php array_push($categoriesFound,$product->category->name);?>
+         @endforeach
+      @elseif( isset($category) )
+         <!-- Category Buttons -->
+         <span style="font-size:1.4em;">Refine your search:</span>
+         @foreach ($products as $product)
+            @if( $product->category->parent->name == $category->name )
+               @if(!in_array($product->category->name,$categoriesFound))
+                  <a role="button" class="btn btn-secondary" href="{{$product->category->id}}">{{$product->category->name}}</a>
+               @endif
+               <?php array_push($categoriesFound,$product->category->name);?>
+            @endif
+         @endforeach
+      @endif
     </div>
-    <!-- End Row -->
+    <!-- End Row Category Search-->
 
     <!-- Row Fluid -->
-    <div class="row-fluid">
+    <div class="row-fluid pt-3">
     	<!-- CARD COLUMNNS -->
     	<div class="card-columns">
     		@foreach ($products as $product)
-    		<!-- Card -->
-    		<div class="card mb-3">
-    			<a href="/ads/{{ $product->id }}" ><img class="card-img-top img-fluid" width="100%" src="{{ asset($product->image) }}" alt="Post Image"></a>
-    			<div class="card-block p-3">
-    				<a id="cardTitle" href="/ads/{{ $product->id }}"><span class="card-title h4 text-justify">{{ $product->title }}</span></a>
-    				<p class="card-text mb-2">{{ $product->description }}</p>
-    				<footer class="text-right">
-    					<small class="text-muted">{{ date('M-jS', strtotime($product->created_at)) }}</small><br/>
-    					<a href="/categories/{{$product->category->id}}"><small class="badge badge-pill badge-info">{{ $product->category->name}}</small></a>
-    					<small class="badge badge-pill badge-success">$ {{ $product->price }}</small>
-    				</footer>
-    			</div>
-    		</div>
-    		<!-- End Card -->
+            @if( isset($category) && ( ($product->category->name == $category->name) || (isset($product->category->parent) && $product->category->parent->name == $category->name) || (isset($product->category->parent->parent) && $product->category->parent->parent->name == $category->name) ) )
```

Comment:
This line is super messy, maybe you should make an array of all the category ids in the controller and pass it to the view and then simply use in_array.

http://php.net/manual/en/function.in-array.php

Reply:
hmm... i agree. However it is functional and there is alot of work to do. I will create an issue and resolve it in the next sprint.

Reply:
alright